editor.planx.uk/src/@planx/components/List/utils.tsx

```diff
@@ -1,24 +1,40 @@
-import { QuestionField, Schema, UserResponse } from "./model";
+import React from "react";
+
+import { Field, UserResponse } from "./model";
 
 /**
- * In the case of "question" fields, ensure the displayed value reflects option "text", rather than "val" as recorded in passport
+ * In the case of "question" and "checklist" fields, ensure the displayed value reflects option "text", rather than "val" as recorded in passport
  * @param value - the `val` or `text` of an Option defined in the schema's fields
- * @param schema - the Schema object
- * @returns string - the `text` for the given value `val`, or the original value
+ * @param field - the Field object
+ * @returns string | React.JSX.Element - the `text` for the given value `val`, or the original value
  */
-export function formatSchemaDisplayValue(value: string, schema: Schema) {
-  const questionFields = schema.fields.filter(
-    (field) => field.type === "question",
-  ) as QuestionField[];
-  const matchingField = questionFields?.find((field) =>
-    field.data.options.some((option) => option.data.val === value),
-  );
-  const matchingOption = matchingField?.data.options.find(
-    (option) => option.data.val === value,
-  );
-
-  // If we found a "val" match, return its text, else just return the value as passed in
-  return matchingOption?.data?.text || value;
+export function formatSchemaDisplayValue(
+  value: string | string[],
+  field: Field,
```

Comment:
Passing in the `Field` as opposed to the `Schema` allows us to simplify how we find the matching fields.